engine/runtime/src/main/java/org/enso/EngineRunnerBootLoader.java

```diff
@@ -0,0 +1,56 @@
+package org.enso;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+
+/**
+ * {@code runner.jar} is a fat jar containing all the dependencies for engine-runner, however, it
+ * cannot be put on module-path, neither class-path, because it cannot be used in conjunction with
+ * {@code runtime.jar} fat jar. For now, this class is a workaround that just tries to invoke {@link
+ * org.enso.runner.Main.main} from {@code runner.jar} using a custom class loader.
+ */
+public final class EngineRunnerBootLoader {
+  private EngineRunnerBootLoader() {}
+
+  private static final String defaultRunnerJar = "runner/runner.jar";
+
+  private static final ClassLoader loader;
+
+  static {
+    var ensoRunnerProp = System.getProperty("enso.runner");
+    var runnerJarPath =
+        ensoRunnerProp != null ? Path.of(ensoRunnerProp) : getDefaultRunnerJarPath();
+    if (!runnerJarPath.toFile().exists()) {
+      throw new IllegalStateException("Cannot find runner fat jar at " + runnerJarPath);
+    }
+    URL url;
+    try {
+      url = runnerJarPath.toUri().toURL();
+    } catch (MalformedURLException e) {
+      throw new IllegalStateException(e);
+    }
+    var parentLoader = ClassLoader.getSystemClassLoader();
+    loader = new URLClassLoader("RunnerBootLoader", new URL[] {url}, parentLoader);
```

Comment:
A core principle of https://github.com/enso-org/enso/pull/7991/commits/2cd617095ea52df7b6f1ab821b52805dc9323ccb#r1368521558 was ability to ignore classes coming from "module layer". Why has that been removed?

We want `runner.jar` to be a fat JAR and just ignore anything that we currently have on module path. Btw. even simpler way to do it is:

```suggestion
    loader = new URLClassLoader("RunnerBootLoader", new URL[] {url}, null);
```